apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -851,6 +858,58 @@ impl Plugin for Telemetry {
             .boxed()
     }
 
+    fn http_client_service(
```

Comment:
I might be wrong but I'm wondering here what happens if I try to count for `http.client.request.duration` and it's a subgraph and it also contains connector attributes. For a subgraph we will pass both in subgraph service and http client service. Which means we could measure the value twice and so increment twice instead of just one. I think with the idea of having everything in subgraph service we should just move the subgraph logics here. During our discussion I thought we have a connector_service or so that would allow us to not move the current logics of subgraph into http_client_service. So I think we have 2 options, either we put this in a connector service or we migrate everything into the http_client_service (the latter is a more long term strategy I think). But that means we will have to move some stuffs into the context to make sure we have access to the same data we need for current subgraph selectors

Reply:
This is the reason for the line:

```rust
.contains_key(CONNECTOR_INFO_CONTEXT_KEY)
```

It will currently only invoke the connector http `on_request` and `on_response` for connectors, not subgraphs, so it will only be called once.

Reply:
Oh ok I missed that,in the future we will be able to get rid of that if we move everything in the http_client_service